### ISSUE TITLE
Headings structure on analytics reports pages present issues for keyboard navigation & screen-readers

### ISSUE THREAD

**Issue by joelclimbsthings** [woocommerce/woocommerce-admin#6208]
**Describe the bug**
A common feature with screen-readers is the ability to navigate via the hierarchy of headings on the page. The way these are structured on the Reports pages may present some difficulty in navigation. 

One concrete example: the report tables do not include a heading at all, making them unreachable when navigating via this method. They do include elements that are clearing headings visually, but not semantically. This should be changed to be a semantic heading. 

**To Reproduce**

- Enable a screen reader that supports heading navigation (ie NVDA)
- Navigate to an analytics page with a data table
- Utilize heading navigation via the keyboard to explore the page, and notice that the data-table is unreachable
- View the source, and notice that the headings on the data table are not semantic heading elements

**Screenshots:**

![image](https://user-images.githubusercontent.com/444632/106068025-b4e1aa80-60b4-11eb-9c7b-cc86849ac7be.png)

**Expected behavior**
The headings on the report tables should be semantic heading elements
